src/shared_memory_manager.cc

```diff
@@ -264,6 +264,46 @@ OpenCudaIPCRegion(
 
   return nullptr;
 }
+
+TRITONSERVER_Error*
+GetCudaSharedMemoryRegionSize(CUdeviceptr data_ptr, size_t& shm_region_size)
+{
+  CUdeviceptr* base = nullptr;
+  CUresult result = cuMemGetAddressRange(base, &shm_region_size, data_ptr);
+  if (result != CUDA_SUCCESS) {
+    const char* errorString;
+    if (cuGetErrorString(result, &errorString) != CUDA_SUCCESS) {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL, "Failed to get CUDA error string");
+    }
+    return TRITONSERVER_ErrorNew(
+        TRITONSERVER_ERROR_INTERNAL,
+        std::string(
+            "Failed to get CUDA address range: " + std::string(errorString))
```

Comment:
do these errors go back to the client?

Reply:
if so - I recommend logging the detailed error but not passing the detailed error back to client / user of api

Reply:
Good catch. Yes, the errors will be passed back to the client. Changed it to just pass a less informative error to the client.